internal/rbd/rbd_util.go

```diff
@@ -651,74 +624,77 @@ func (rv *rbdVolume) ensureImageCleanup(ctx context.Context) error {
 		if val.Name == rv.RbdImageName {
 			rv.ImageID = val.Id
 
-			return trashRemoveImage(ctx, rv, rv.conn.Creds)
+			return rv.trashRemoveImage(ctx)
 		}
 	}
 
 	return nil
 }
 
 // deleteImage deletes a ceph image with provision and volume options.
-func deleteImage(ctx context.Context, pOpts *rbdVolume, cr *util.Credentials) error {
-	image := pOpts.RbdImageName
+func (rv *rbdVolume) deleteImage(ctx context.Context) error {
+	image := rv.RbdImageName
 
-	log.DebugLog(ctx, "rbd: delete %s using mon %s, pool %s", image, pOpts.Monitors, pOpts.Pool)
+	log.DebugLog(ctx, "rbd: delete %s using mon %s, pool %s", image, rv.Monitors, rv.Pool)
 
 	// Support deleting the older rbd images whose imageID is not stored in omap
-	err := pOpts.getImageID()
+	err := rv.getImageID()
 	if err != nil {
 		return err
 	}
 
-	if pOpts.isEncrypted() {
-		log.DebugLog(ctx, "rbd: going to remove DEK for %q", pOpts)
-		if err = pOpts.encryption.RemoveDEK(pOpts.VolID); err != nil {
-			log.WarningLog(ctx, "failed to clean the passphrase for volume %s: %s", pOpts.VolID, err)
+	if rv.isEncrypted() {
+		log.DebugLog(ctx, "rbd: going to remove DEK for %q", rv)
+		if err = rv.encryption.RemoveDEK(rv.VolID); err != nil {
+			log.WarningLog(ctx, "failed to clean the passphrase for volume %s: %s", rv.VolID, err)
 		}
 	}
 
-	err = pOpts.openIoctx()
+	err = rv.openIoctx()
 	if err != nil {
 		return err
 	}
 
-	rbdImage := librbd.GetImage(pOpts.ioctx, image)
+	rbdImage := librbd.GetImage(rv.ioctx, image)
 	err = rbdImage.Trash(0)
 	if err != nil {
-		log.ErrorLog(ctx, "failed to delete rbd image: %s, error: %v", pOpts, err)
+		log.ErrorLog(ctx, "failed to delete rbd image: %s, error: %v", rv, err)
 
 		return err
 	}
 
-	return trashRemoveImage(ctx, pOpts, cr)
+	return rv.trashRemoveImage(ctx)
 }
 
 // trashRemoveImage adds a task to trash remove an image using ceph manager if supported,
 // otherwise removes the image from trash.
-func trashRemoveImage(ctx context.Context, pOpts *rbdVolume, cr *util.Credentials) error {
+func (rv *rbdVolume) trashRemoveImage(ctx context.Context) error {
```

Comment:
same here make this method of `rbdImage` struct?

Reply:
Still pending? if there is a reason not to do that, document it.

Reply:
Opened issue to track it https://github.com/ceph/ceph-csi/issues/2742